<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Annotating the results of Array-based DNA methylation data â€¢ BasicR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Annotating the results of Array-based DNA methylation data"><meta property="og:description" content="BasicR"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BasicR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/assignment.html">Assignment</a>
    </li>
    <li>
      <a href="../articles/basic_functions.html">Basic functions</a>
    </li>
    <li>
      <a href="../articles/control_structures.html">Control structures</a>
    </li>
    <li>
      <a href="../articles/data_cleaning.html">Data cleaning with Tidyverse</a>
    </li>
    <li>
      <a href="../articles/data_types_exercises.html">Data types, exercises</a>
    </li>
    <li>
      <a href="../articles/data_types_exercises_solutions.html">Data types, solutions</a>
    </li>
    <li>
      <a href="../articles/enrichment.html">Enrichment analysis</a>
    </li>
    <li>
      <a href="../articles/exercises_control_exercises.html">Control structures, exercises</a>
    </li>
    <li>
      <a href="../articles/exercises_control_solutions.html">Control structures, solutions</a>
    </li>
    <li>
      <a href="../articles/functions_exercises.html">Functions, exercises</a>
    </li>
    <li>
      <a href="../articles/functions_exercises_solutions.html">Functions, solutions</a>
    </li>
    <li>
      <a href="../articles/intro.html">Course introduction</a>
    </li>
    <li>
      <a href="../articles/methylation_analysisRmd.html">Annotating the results of Array-based DNA methylation data</a>
    </li>
    <li>
      <a href="../articles/references.html">References</a>
    </li>
    <li>
      <a href="../articles/reproducible.html">Reproducible programming</a>
    </li>
    <li>
      <a href="../articles/reproducible_solutions.html">Reproducible programming, exercises</a>
    </li>
    <li>
      <a href="../articles/resources.html">Resources</a>
    </li>
    <li>
      <a href="../articles/R_data_types.html">R Data types</a>
    </li>
    <li>
      <a href="../articles/r_functions.html">Functions, useful packages</a>
    </li>
    <li>
      <a href="../articles/S4_classes.html">S4 classes</a>
    </li>
    <li>
      <a href="../articles/S4_classes_exercises.html">S4 classes, exercises</a>
    </li>
    <li>
      <a href="../articles/set_up_git.html">Set up git</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/CompEpigen/BasicR" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>





<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Annotating the results of Array-based DNA
methylation data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/CompEpigen/BasicR/blob/HEAD/vignettes/methylation_analysisRmd.Rmd" class="external-link"><code>vignettes/methylation_analysisRmd.Rmd</code></a></small>
      <div class="hidden name"><code>methylation_analysisRmd.Rmd</code></div>

    </div>

    
    
<div id="visualization" class="section level1">
<h1>Visualization</h1>
<ol style="list-style-type: decimal">
<li>Load the data to the UCSC (IGV) browser to visualize it.</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Use the bed files from the tracks and tables folder</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Check for the gene LIG4</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Add additional tracks, like functional regions from ENCODE, TFBSs,
etc</li>
</ol>
</div>
<div id="location" class="section level1">
<h1>Location</h1>
<ol style="list-style-type: decimal">
<li>Annotate regions using the annotatr package</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#file &lt;- fread(file=&quot;data/diffMethTable_site_cmp1.csv&quot;, sep = &quot;,&quot;)</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#bg &lt;- copy(file)</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#bg &lt;- bg[,.(Chromosome, Start, End=Start+1, cgid)]</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#fwrite(bg, row.names=F, file = &quot;rnb_run_pilot/differential_methylation_data/background.txt&quot;, quote = F, sep=&quot;\t&quot;,  col.names = F)</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co">#hyper &lt;- file[diffmeth.p.val&lt;0.0001 &amp; mean.diff&gt;0.3,]</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co">#great_hyper &lt;- hyper[,.(Chromosome, Start, End=Start+1, cgid)]</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co">#hypo &lt;- file[diffmeth.p.val&lt;10e-14 &amp; mean.diff&lt; -0.7,]</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co">#great_hypo &lt;- hypo[,.(Chromosome, Start, End=Start+1, cgid)]</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="co">#fwrite(file, row.names=F, file = paste0(folder, &quot;diffMethTable_significant01_cmp_&quot;, i, &quot;.txt&quot;), quote = F, sep=&quot;\t&quot;)</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="co">#fwrite(great_hyper, row.names=F, file = &quot;rnb_run_pilot/differential_methylation_data/great_significant_hyper.txt&quot;, quote = F, sep=&quot;\t&quot;, col.names = F)</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="co">#fwrite(great_hypo, row.names=F, file = &quot;rnb_run_pilot/differential_methylation_data/great_significant_hypo.txt&quot;, quote = F, sep=&quot;\t&quot;, col.names = F)</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>data[[<span class="st">&quot;hypo&quot;</span>]]  <span class="ot">&lt;-</span> <span class="fu">import.bed</span>(<span class="st">&quot;../inst/extdata/great_significant_hypo.txt&quot;</span>)</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>data[[<span class="st">&quot;hyper&quot;</span>]]<span class="ot">&lt;-</span> <span class="fu">import.bed</span>(<span class="st">&quot;../inst/extdata/great_significant_hyper.txt&quot;</span>)</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>annots <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;hg19_Hepg2-chromatin&#39;</span>)</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>annots_gr <span class="ot">=</span> <span class="fu">build_annotations</span>(<span class="at">genome =</span> <span class="st">&#39;hg19&#39;</span>, <span class="at">annotations =</span> annots)</span></code></pre></div>
<pre><code>## Building hmms...</code></pre>
<pre><code>## Downloading chromHMM track for Hepg2</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (dataset <span class="cf">in</span> data){</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">genome</span>(dataset) <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;hg19&quot;</span>, <span class="fu">length</span>(<span class="fu">genome</span>(dataset)))</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">annotate_regions</span>(dataset, <span class="at">annotations=</span>annots_gr, <span class="at">minoverlap =</span> 1L, <span class="at">ignore.strand =</span> <span class="cn">TRUE</span>, <span class="at">quiet =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>rnd_annots <span class="ot">=</span> <span class="fu">annotate_regions</span>(<span class="at">regions =</span> <span class="fu">randomize_regions</span>(dataset),<span class="at">annotations =</span> annots_gr,<span class="at">ignore.strand =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>p_annots <span class="ot">=</span> <span class="fu">plot_annotation</span>(<span class="at">annotated_regions =</span> result, <span class="at">annotated_random =</span> rnd_annots)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p_annots)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## Annotating...</code></pre>
<pre><code>## Randomizing regions...</code></pre>
<pre><code>## Annotating...
## Annotating...</code></pre>
<pre><code>## Randomizing regions...</code></pre>
<pre><code>## Annotating...</code></pre>
<p><img src="methylation_analysisRmd_files/figure-html/unnamed-chunk-3-1.png" width="700" /><img src="methylation_analysisRmd_files/figure-html/unnamed-chunk-3-2.png" width="700" /></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>background <span class="ot">&lt;-</span>  <span class="fu">import.bed</span>(<span class="st">&quot;../inst/extdata/background.txt&quot;</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">genome</span>(background) <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;hg19&quot;</span>, <span class="fu">length</span>(<span class="fu">genome</span>(background)))</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (dataset <span class="cf">in</span> <span class="fu">names</span>(data)){</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="fu">genome</span>(data[[dataset]]) <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;hg19&quot;</span>, <span class="fu">length</span>(<span class="fu">genome</span>(data[[dataset]])))</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>background2 <span class="ot">&lt;-</span> background[<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(background), <span class="fu">length</span>(data[[dataset]])),]</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">annotate_regions</span>(data[[dataset]], <span class="at">annotations=</span>annots_gr, <span class="at">minoverlap =</span> 1L, <span class="at">ignore.strand =</span> <span class="cn">TRUE</span>, <span class="at">quiet =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>rnd_annots <span class="ot">=</span> <span class="fu">annotate_regions</span>(<span class="at">regions =</span> background2,<span class="at">annotations =</span> annots_gr,<span class="at">ignore.strand =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>p_annots <span class="ot">=</span> <span class="fu">plot_annotation</span>(<span class="at">annotated_regions =</span> result, <span class="at">annotated_random =</span> rnd_annots)</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p_annots<span class="sc">+</span><span class="fu">ggtitle</span>(dataset))</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## Annotating...
## Annotating...
## Annotating...
## Annotating...</code></pre>
<p><img src="methylation_analysisRmd_files/figure-html/unnamed-chunk-3-3.png" width="700" /><img src="methylation_analysisRmd_files/figure-html/unnamed-chunk-3-4.png" width="700" /></p>
<ol start="2" style="list-style-type: decimal">
<li>Try different annotations</li>
</ol>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">builtin_annotations</span>()</span></code></pre></div>
<pre><code>##   [1] &quot;dm3_genes_1to5kb&quot;                     
##   [2] &quot;dm6_genes_1to5kb&quot;                     
##   [3] &quot;galGal5_genes_1to5kb&quot;                 
##   [4] &quot;hg19_genes_1to5kb&quot;                    
##   [5] &quot;hg38_genes_1to5kb&quot;                    
##   [6] &quot;mm9_genes_1to5kb&quot;                     
##   [7] &quot;mm10_genes_1to5kb&quot;                    
##   [8] &quot;rn4_genes_1to5kb&quot;                     
##   [9] &quot;rn5_genes_1to5kb&quot;                     
##  [10] &quot;rn6_genes_1to5kb&quot;                     
##  [11] &quot;dm3_genes_promoters&quot;                  
##  [12] &quot;dm6_genes_promoters&quot;                  
##  [13] &quot;galGal5_genes_promoters&quot;              
##  [14] &quot;hg19_genes_promoters&quot;                 
##  [15] &quot;hg38_genes_promoters&quot;                 
##  [16] &quot;mm9_genes_promoters&quot;                  
##  [17] &quot;mm10_genes_promoters&quot;                 
##  [18] &quot;rn4_genes_promoters&quot;                  
##  [19] &quot;rn5_genes_promoters&quot;                  
##  [20] &quot;rn6_genes_promoters&quot;                  
##  [21] &quot;dm3_genes_cds&quot;                        
##  [22] &quot;dm6_genes_cds&quot;                        
##  [23] &quot;galGal5_genes_cds&quot;                    
##  [24] &quot;hg19_genes_cds&quot;                       
##  [25] &quot;hg38_genes_cds&quot;                       
##  [26] &quot;mm9_genes_cds&quot;                        
##  [27] &quot;mm10_genes_cds&quot;                       
##  [28] &quot;rn4_genes_cds&quot;                        
##  [29] &quot;rn5_genes_cds&quot;                        
##  [30] &quot;rn6_genes_cds&quot;                        
##  [31] &quot;dm3_genes_5UTRs&quot;                      
##  [32] &quot;dm6_genes_5UTRs&quot;                      
##  [33] &quot;galGal5_genes_5UTRs&quot;                  
##  [34] &quot;hg19_genes_5UTRs&quot;                     
##  [35] &quot;hg38_genes_5UTRs&quot;                     
##  [36] &quot;mm9_genes_5UTRs&quot;                      
##  [37] &quot;mm10_genes_5UTRs&quot;                     
##  [38] &quot;rn4_genes_5UTRs&quot;                      
##  [39] &quot;rn5_genes_5UTRs&quot;                      
##  [40] &quot;rn6_genes_5UTRs&quot;                      
##  [41] &quot;dm3_genes_exons&quot;                      
##  [42] &quot;dm6_genes_exons&quot;                      
##  [43] &quot;galGal5_genes_exons&quot;                  
##  [44] &quot;hg19_genes_exons&quot;                     
##  [45] &quot;hg38_genes_exons&quot;                     
##  [46] &quot;mm9_genes_exons&quot;                      
##  [47] &quot;mm10_genes_exons&quot;                     
##  [48] &quot;rn4_genes_exons&quot;                      
##  [49] &quot;rn5_genes_exons&quot;                      
##  [50] &quot;rn6_genes_exons&quot;                      
##  [51] &quot;dm3_genes_firstexons&quot;                 
##  [52] &quot;dm6_genes_firstexons&quot;                 
##  [53] &quot;galGal5_genes_firstexons&quot;             
##  [54] &quot;hg19_genes_firstexons&quot;                
##  [55] &quot;hg38_genes_firstexons&quot;                
##  [56] &quot;mm9_genes_firstexons&quot;                 
##  [57] &quot;mm10_genes_firstexons&quot;                
##  [58] &quot;rn4_genes_firstexons&quot;                 
##  [59] &quot;rn5_genes_firstexons&quot;                 
##  [60] &quot;rn6_genes_firstexons&quot;                 
##  [61] &quot;dm3_genes_introns&quot;                    
##  [62] &quot;dm6_genes_introns&quot;                    
##  [63] &quot;galGal5_genes_introns&quot;                
##  [64] &quot;hg19_genes_introns&quot;                   
##  [65] &quot;hg38_genes_introns&quot;                   
##  [66] &quot;mm9_genes_introns&quot;                    
##  [67] &quot;mm10_genes_introns&quot;                   
##  [68] &quot;rn4_genes_introns&quot;                    
##  [69] &quot;rn5_genes_introns&quot;                    
##  [70] &quot;rn6_genes_introns&quot;                    
##  [71] &quot;dm3_genes_intronexonboundaries&quot;       
##  [72] &quot;dm6_genes_intronexonboundaries&quot;       
##  [73] &quot;galGal5_genes_intronexonboundaries&quot;   
##  [74] &quot;hg19_genes_intronexonboundaries&quot;      
##  [75] &quot;hg38_genes_intronexonboundaries&quot;      
##  [76] &quot;mm9_genes_intronexonboundaries&quot;       
##  [77] &quot;mm10_genes_intronexonboundaries&quot;      
##  [78] &quot;rn4_genes_intronexonboundaries&quot;       
##  [79] &quot;rn5_genes_intronexonboundaries&quot;       
##  [80] &quot;rn6_genes_intronexonboundaries&quot;       
##  [81] &quot;dm3_genes_exonintronboundaries&quot;       
##  [82] &quot;dm6_genes_exonintronboundaries&quot;       
##  [83] &quot;galGal5_genes_exonintronboundaries&quot;   
##  [84] &quot;hg19_genes_exonintronboundaries&quot;      
##  [85] &quot;hg38_genes_exonintronboundaries&quot;      
##  [86] &quot;mm9_genes_exonintronboundaries&quot;       
##  [87] &quot;mm10_genes_exonintronboundaries&quot;      
##  [88] &quot;rn4_genes_exonintronboundaries&quot;       
##  [89] &quot;rn5_genes_exonintronboundaries&quot;       
##  [90] &quot;rn6_genes_exonintronboundaries&quot;       
##  [91] &quot;dm3_genes_3UTRs&quot;                      
##  [92] &quot;dm6_genes_3UTRs&quot;                      
##  [93] &quot;galGal5_genes_3UTRs&quot;                  
##  [94] &quot;hg19_genes_3UTRs&quot;                     
##  [95] &quot;hg38_genes_3UTRs&quot;                     
##  [96] &quot;mm9_genes_3UTRs&quot;                      
##  [97] &quot;mm10_genes_3UTRs&quot;                     
##  [98] &quot;rn4_genes_3UTRs&quot;                      
##  [99] &quot;rn5_genes_3UTRs&quot;                      
## [100] &quot;rn6_genes_3UTRs&quot;                      
## [101] &quot;dm3_genes_intergenic&quot;                 
## [102] &quot;dm6_genes_intergenic&quot;                 
## [103] &quot;galGal5_genes_intergenic&quot;             
## [104] &quot;hg19_genes_intergenic&quot;                
## [105] &quot;hg38_genes_intergenic&quot;                
## [106] &quot;mm9_genes_intergenic&quot;                 
## [107] &quot;mm10_genes_intergenic&quot;                
## [108] &quot;rn4_genes_intergenic&quot;                 
## [109] &quot;rn5_genes_intergenic&quot;                 
## [110] &quot;rn6_genes_intergenic&quot;                 
## [111] &quot;galGal5_cpg_islands&quot;                  
## [112] &quot;hg19_cpg_islands&quot;                     
## [113] &quot;hg38_cpg_islands&quot;                     
## [114] &quot;mm9_cpg_islands&quot;                      
## [115] &quot;mm10_cpg_islands&quot;                     
## [116] &quot;rn4_cpg_islands&quot;                      
## [117] &quot;rn5_cpg_islands&quot;                      
## [118] &quot;rn6_cpg_islands&quot;                      
## [119] &quot;galGal5_cpg_shores&quot;                   
## [120] &quot;hg19_cpg_shores&quot;                      
## [121] &quot;hg38_cpg_shores&quot;                      
## [122] &quot;mm9_cpg_shores&quot;                       
## [123] &quot;mm10_cpg_shores&quot;                      
## [124] &quot;rn4_cpg_shores&quot;                       
## [125] &quot;rn5_cpg_shores&quot;                       
## [126] &quot;rn6_cpg_shores&quot;                       
## [127] &quot;galGal5_cpg_shelves&quot;                  
## [128] &quot;hg19_cpg_shelves&quot;                     
## [129] &quot;hg38_cpg_shelves&quot;                     
## [130] &quot;mm9_cpg_shelves&quot;                      
## [131] &quot;mm10_cpg_shelves&quot;                     
## [132] &quot;rn4_cpg_shelves&quot;                      
## [133] &quot;rn5_cpg_shelves&quot;                      
## [134] &quot;rn6_cpg_shelves&quot;                      
## [135] &quot;galGal5_cpg_inter&quot;                    
## [136] &quot;hg19_cpg_inter&quot;                       
## [137] &quot;hg38_cpg_inter&quot;                       
## [138] &quot;mm9_cpg_inter&quot;                        
## [139] &quot;mm10_cpg_inter&quot;                       
## [140] &quot;rn4_cpg_inter&quot;                        
## [141] &quot;rn5_cpg_inter&quot;                        
## [142] &quot;rn6_cpg_inter&quot;                        
## [143] &quot;hg19_chromatin_Gm12878-ActivePromoter&quot;
## [144] &quot;hg19_chromatin_H1hesc-ActivePromoter&quot; 
## [145] &quot;hg19_chromatin_Hepg2-ActivePromoter&quot;  
## [146] &quot;hg19_chromatin_Hmec-ActivePromoter&quot;   
## [147] &quot;hg19_chromatin_Hsmm-ActivePromoter&quot;   
## [148] &quot;hg19_chromatin_Huvec-ActivePromoter&quot;  
## [149] &quot;hg19_chromatin_K562-ActivePromoter&quot;   
## [150] &quot;hg19_chromatin_Nhek-ActivePromoter&quot;   
## [151] &quot;hg19_chromatin_Nhlf-ActivePromoter&quot;   
## [152] &quot;hg19_chromatin_Gm12878-WeakPromoter&quot;  
## [153] &quot;hg19_chromatin_H1hesc-WeakPromoter&quot;   
## [154] &quot;hg19_chromatin_Hepg2-WeakPromoter&quot;    
## [155] &quot;hg19_chromatin_Hmec-WeakPromoter&quot;     
## [156] &quot;hg19_chromatin_Hsmm-WeakPromoter&quot;     
## [157] &quot;hg19_chromatin_Huvec-WeakPromoter&quot;    
## [158] &quot;hg19_chromatin_K562-WeakPromoter&quot;     
## [159] &quot;hg19_chromatin_Nhek-WeakPromoter&quot;     
## [160] &quot;hg19_chromatin_Nhlf-WeakPromoter&quot;     
## [161] &quot;hg19_chromatin_Gm12878-PoisedPromoter&quot;
## [162] &quot;hg19_chromatin_H1hesc-PoisedPromoter&quot; 
## [163] &quot;hg19_chromatin_Hepg2-PoisedPromoter&quot;  
## [164] &quot;hg19_chromatin_Hmec-PoisedPromoter&quot;   
## [165] &quot;hg19_chromatin_Hsmm-PoisedPromoter&quot;   
## [166] &quot;hg19_chromatin_Huvec-PoisedPromoter&quot;  
## [167] &quot;hg19_chromatin_K562-PoisedPromoter&quot;   
## [168] &quot;hg19_chromatin_Nhek-PoisedPromoter&quot;   
## [169] &quot;hg19_chromatin_Nhlf-PoisedPromoter&quot;   
## [170] &quot;hg19_chromatin_Gm12878-StrongEnhancer&quot;
## [171] &quot;hg19_chromatin_H1hesc-StrongEnhancer&quot; 
## [172] &quot;hg19_chromatin_Hepg2-StrongEnhancer&quot;  
## [173] &quot;hg19_chromatin_Hmec-StrongEnhancer&quot;   
## [174] &quot;hg19_chromatin_Hsmm-StrongEnhancer&quot;   
## [175] &quot;hg19_chromatin_Huvec-StrongEnhancer&quot;  
## [176] &quot;hg19_chromatin_K562-StrongEnhancer&quot;   
## [177] &quot;hg19_chromatin_Nhek-StrongEnhancer&quot;   
## [178] &quot;hg19_chromatin_Nhlf-StrongEnhancer&quot;   
## [179] &quot;hg19_chromatin_Gm12878-WeakEnhancer&quot;  
## [180] &quot;hg19_chromatin_H1hesc-WeakEnhancer&quot;   
## [181] &quot;hg19_chromatin_Hepg2-WeakEnhancer&quot;    
## [182] &quot;hg19_chromatin_Hmec-WeakEnhancer&quot;     
## [183] &quot;hg19_chromatin_Hsmm-WeakEnhancer&quot;     
## [184] &quot;hg19_chromatin_Huvec-WeakEnhancer&quot;    
## [185] &quot;hg19_chromatin_K562-WeakEnhancer&quot;     
## [186] &quot;hg19_chromatin_Nhek-WeakEnhancer&quot;     
## [187] &quot;hg19_chromatin_Nhlf-WeakEnhancer&quot;     
## [188] &quot;hg19_chromatin_Gm12878-Insulator&quot;     
## [189] &quot;hg19_chromatin_H1hesc-Insulator&quot;      
## [190] &quot;hg19_chromatin_Hepg2-Insulator&quot;       
## [191] &quot;hg19_chromatin_Hmec-Insulator&quot;        
## [192] &quot;hg19_chromatin_Hsmm-Insulator&quot;        
## [193] &quot;hg19_chromatin_Huvec-Insulator&quot;       
## [194] &quot;hg19_chromatin_K562-Insulator&quot;        
## [195] &quot;hg19_chromatin_Nhek-Insulator&quot;        
## [196] &quot;hg19_chromatin_Nhlf-Insulator&quot;        
## [197] &quot;hg19_chromatin_Gm12878-TxnTransition&quot; 
## [198] &quot;hg19_chromatin_H1hesc-TxnTransition&quot;  
## [199] &quot;hg19_chromatin_Hepg2-TxnTransition&quot;   
## [200] &quot;hg19_chromatin_Hmec-TxnTransition&quot;    
## [201] &quot;hg19_chromatin_Hsmm-TxnTransition&quot;    
## [202] &quot;hg19_chromatin_Huvec-TxnTransition&quot;   
## [203] &quot;hg19_chromatin_K562-TxnTransition&quot;    
## [204] &quot;hg19_chromatin_Nhek-TxnTransition&quot;    
## [205] &quot;hg19_chromatin_Nhlf-TxnTransition&quot;    
## [206] &quot;hg19_chromatin_Gm12878-TxnElongation&quot; 
## [207] &quot;hg19_chromatin_H1hesc-TxnElongation&quot;  
## [208] &quot;hg19_chromatin_Hepg2-TxnElongation&quot;   
## [209] &quot;hg19_chromatin_Hmec-TxnElongation&quot;    
## [210] &quot;hg19_chromatin_Hsmm-TxnElongation&quot;    
## [211] &quot;hg19_chromatin_Huvec-TxnElongation&quot;   
## [212] &quot;hg19_chromatin_K562-TxnElongation&quot;    
## [213] &quot;hg19_chromatin_Nhek-TxnElongation&quot;    
## [214] &quot;hg19_chromatin_Nhlf-TxnElongation&quot;    
## [215] &quot;hg19_chromatin_Gm12878-WeakTxn&quot;       
## [216] &quot;hg19_chromatin_H1hesc-WeakTxn&quot;        
## [217] &quot;hg19_chromatin_Hepg2-WeakTxn&quot;         
## [218] &quot;hg19_chromatin_Hmec-WeakTxn&quot;          
## [219] &quot;hg19_chromatin_Hsmm-WeakTxn&quot;          
## [220] &quot;hg19_chromatin_Huvec-WeakTxn&quot;         
## [221] &quot;hg19_chromatin_K562-WeakTxn&quot;          
## [222] &quot;hg19_chromatin_Nhek-WeakTxn&quot;          
## [223] &quot;hg19_chromatin_Nhlf-WeakTxn&quot;          
## [224] &quot;hg19_chromatin_Gm12878-Repressed&quot;     
## [225] &quot;hg19_chromatin_H1hesc-Repressed&quot;      
## [226] &quot;hg19_chromatin_Hepg2-Repressed&quot;       
## [227] &quot;hg19_chromatin_Hmec-Repressed&quot;        
## [228] &quot;hg19_chromatin_Hsmm-Repressed&quot;        
## [229] &quot;hg19_chromatin_Huvec-Repressed&quot;       
## [230] &quot;hg19_chromatin_K562-Repressed&quot;        
## [231] &quot;hg19_chromatin_Nhek-Repressed&quot;        
## [232] &quot;hg19_chromatin_Nhlf-Repressed&quot;        
## [233] &quot;hg19_chromatin_Gm12878-Heterochrom/lo&quot;
## [234] &quot;hg19_chromatin_H1hesc-Heterochrom/lo&quot; 
## [235] &quot;hg19_chromatin_Hepg2-Heterochrom/lo&quot;  
## [236] &quot;hg19_chromatin_Hmec-Heterochrom/lo&quot;   
## [237] &quot;hg19_chromatin_Hsmm-Heterochrom/lo&quot;   
## [238] &quot;hg19_chromatin_Huvec-Heterochrom/lo&quot;  
## [239] &quot;hg19_chromatin_K562-Heterochrom/lo&quot;   
## [240] &quot;hg19_chromatin_Nhek-Heterochrom/lo&quot;   
## [241] &quot;hg19_chromatin_Nhlf-Heterochrom/lo&quot;   
## [242] &quot;hg19_chromatin_Gm12878-Repetitive/CNV&quot;
## [243] &quot;hg19_chromatin_H1hesc-Repetitive/CNV&quot; 
## [244] &quot;hg19_chromatin_Hepg2-Repetitive/CNV&quot;  
## [245] &quot;hg19_chromatin_Hmec-Repetitive/CNV&quot;   
## [246] &quot;hg19_chromatin_Hsmm-Repetitive/CNV&quot;   
## [247] &quot;hg19_chromatin_Huvec-Repetitive/CNV&quot;  
## [248] &quot;hg19_chromatin_K562-Repetitive/CNV&quot;   
## [249] &quot;hg19_chromatin_Nhek-Repetitive/CNV&quot;   
## [250] &quot;hg19_chromatin_Nhlf-Repetitive/CNV&quot;   
## [251] &quot;hg19_enhancers_fantom&quot;                
## [252] &quot;hg38_enhancers_fantom&quot;                
## [253] &quot;mm9_enhancers_fantom&quot;                 
## [254] &quot;mm10_enhancers_fantom&quot;                
## [255] &quot;hg19_lncrna_gencode&quot;                  
## [256] &quot;hg38_lncrna_gencode&quot;                  
## [257] &quot;mm10_lncrna_gencode&quot;                  
## [258] &quot;dm3_basicgenes&quot;                       
## [259] &quot;dm6_basicgenes&quot;                       
## [260] &quot;galGal5_basicgenes&quot;                   
## [261] &quot;hg19_basicgenes&quot;                      
## [262] &quot;hg38_basicgenes&quot;                      
## [263] &quot;mm9_basicgenes&quot;                       
## [264] &quot;mm10_basicgenes&quot;                      
## [265] &quot;rn4_basicgenes&quot;                       
## [266] &quot;rn5_basicgenes&quot;                       
## [267] &quot;rn6_basicgenes&quot;                       
## [268] &quot;galGal5_cpgs&quot;                         
## [269] &quot;hg19_cpgs&quot;                            
## [270] &quot;hg38_cpgs&quot;                            
## [271] &quot;mm9_cpgs&quot;                             
## [272] &quot;mm10_cpgs&quot;                            
## [273] &quot;rn4_cpgs&quot;                             
## [274] &quot;rn5_cpgs&quot;                             
## [275] &quot;rn6_cpgs&quot;                             
## [276] &quot;hg19_Gm12878-chromatin&quot;               
## [277] &quot;hg19_H1hesc-chromatin&quot;                
## [278] &quot;hg19_Hepg2-chromatin&quot;                 
## [279] &quot;hg19_Hmec-chromatin&quot;                  
## [280] &quot;hg19_Hsmm-chromatin&quot;                  
## [281] &quot;hg19_Huvec-chromatin&quot;                 
## [282] &quot;hg19_K562-chromatin&quot;                  
## [283] &quot;hg19_Nhek-chromatin&quot;                  
## [284] &quot;hg19_Nhlf-chromatin&quot;</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Pathway enrichment analysis</li>
</ol>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TxDb.Hsapiens.UCSC.hg19.knownGene)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">&lt;-</span> TxDb.Hsapiens.UCSC.hg19.knownGene</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (dataset <span class="cf">in</span> <span class="fu">names</span>(data)){</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> <span class="fu">seq2gene</span>(data[[dataset]], <span class="at">tssRegion =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">3000</span>, <span class="dv">3000</span>), <span class="at">flankDistance =</span> <span class="dv">3000</span>, <span class="at">TxDb=</span>txdb)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  kkm <span class="ot">&lt;-</span> <span class="fu">enrichKEGG</span>(<span class="at">gene         =</span> genes,</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">organism     =</span> <span class="st">&#39;human&#39;</span>,</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dotplot</span>(kkm))</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  engo <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene          =</span> genes,</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">ont           =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">pvalueCutoff  =</span> <span class="fl">0.01</span>,</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">readable      =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dotplot</span>(engo))</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>  }</span></code></pre></div>
<pre><code>## Reading KEGG annotation online: &quot;https://rest.kegg.jp/link/hsa/pathway&quot;...</code></pre>
<pre><code>## Warning in utils::download.file(url, quiet = TRUE, method = method, ...): the
## &#39;wininet&#39; method is deprecated for http:// and https:// URLs</code></pre>
<pre><code>## Reading KEGG annotation online: &quot;https://rest.kegg.jp/list/pathway&quot;...</code></pre>
<pre><code>## Warning in utils::download.file(url, quiet = TRUE, method = method, ...): the
## &#39;wininet&#39; method is deprecated for http:// and https:// URLs</code></pre>
<p><img src="methylation_analysisRmd_files/figure-html/unnamed-chunk-5-1.png" width="700" /><img src="methylation_analysisRmd_files/figure-html/unnamed-chunk-5-2.png" width="700" /><img src="methylation_analysisRmd_files/figure-html/unnamed-chunk-5-3.png" width="700" /><img src="methylation_analysisRmd_files/figure-html/unnamed-chunk-5-4.png" width="700" /></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
      </div>

</div>



      <footer><div class="copyright">
  <p></p><p>Developed by Reka Toth, Rajbir Nath Batra.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>
