<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>S4 classes • BasicR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="S4 classes"><meta property="og:description" content="BasicR"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BasicR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/assignment.html">Assignment</a>
    </li>
    <li>
      <a href="../articles/basic_functions.html">Basic functions</a>
    </li>
    <li>
      <a href="../articles/control_structures.html">Control structures</a>
    </li>
    <li>
      <a href="../articles/data_cleaning.html">Data cleaning with Tidyverse</a>
    </li>
    <li>
      <a href="../articles/data_types_exercises.html">Data types, exercises</a>
    </li>
    <li>
      <a href="../articles/data_types_exercises_solutions.html">Data types, solutions</a>
    </li>
    <li>
      <a href="../articles/enrichment.html">Enrichment analysis</a>
    </li>
    <li>
      <a href="../articles/exercises_control_exercises.html">Control structures, exercises</a>
    </li>
    <li>
      <a href="../articles/exercises_control_solutions.html">Control structures, solutions</a>
    </li>
    <li>
      <a href="../articles/functions_exercises.html">Functions, exercises</a>
    </li>
    <li>
      <a href="../articles/functions_exercises_solutions.html">Functions, solutions</a>
    </li>
    <li>
      <a href="../articles/intro.html">Course introduction</a>
    </li>
    <li>
      <a href="../articles/references.html">References</a>
    </li>
    <li>
      <a href="../articles/reproducible.html">Reproducible programming</a>
    </li>
    <li>
      <a href="../articles/reproducible_solutions.html">Reproducible programming, exercises</a>
    </li>
    <li>
      <a href="../articles/resources.html">Resources</a>
    </li>
    <li>
      <a href="../articles/R_data_types.html">R Data types</a>
    </li>
    <li>
      <a href="../articles/r_functions.html">Functions, useful packages</a>
    </li>
    <li>
      <a href="../articles/S4_classes.html">S4 classes</a>
    </li>
    <li>
      <a href="../articles/S4_classes_exercises.html">S4 classes, exercises</a>
    </li>
    <li>
      <a href="../articles/set_up_git.html">Set up git</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/CompEpigen/BasicR" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>





<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>S4 classes</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/CompEpigen/BasicR/blob/HEAD/vignettes/S4_classes.Rmd" class="external-link"><code>vignettes/S4_classes.Rmd</code></a></small>
      <div class="hidden name"><code>S4_classes.Rmd</code></div>

    </div>

    
    
<p>S4 classes are the perfect examples of the OO programming: - stricter
structure - they have representations: a list of slots or arguments by
name and class - the slots have to have the correct type - methods -
they also inherit the methods from the classes of the arguments, but
inheritence is between S4 classes as well. - has to be created with the
new constructor - they have accessor methods as well.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setClass</span>(<span class="st">&quot;Person&quot;</span>, <span class="fu">representation</span>(<span class="at">name =</span> <span class="st">&quot;character&quot;</span>, <span class="at">age =</span> <span class="st">&quot;numeric&quot;</span>))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setClass</span>(<span class="st">&quot;Employee&quot;</span>, <span class="fu">representation</span>(<span class="at">boss =</span> <span class="st">&quot;Person&quot;</span>), <span class="at">contains =</span> <span class="st">&quot;Person&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>person1 <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">&quot;Person&quot;</span>, <span class="at">name=</span><span class="st">&quot;Peter&quot;</span>, <span class="at">age=</span><span class="dv">34</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>boss1 <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">&quot;Employee&quot;</span>, <span class="at">boss=</span>person1, <span class="at">name=</span><span class="st">&quot;Lisa&quot;</span>, <span class="at">age=</span><span class="dv">36</span>)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>person1</span></code></pre></div>
<pre><code>## An object of class &quot;Person&quot;
## Slot &quot;name&quot;:
## [1] &quot;Peter&quot;
## 
## Slot &quot;age&quot;:
## [1] 34</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>boss1</span></code></pre></div>
<pre><code>## An object of class &quot;Employee&quot;
## Slot &quot;boss&quot;:
## An object of class &quot;Person&quot;
## Slot &quot;name&quot;:
## [1] &quot;Peter&quot;
## 
## Slot &quot;age&quot;:
## [1] 34
## 
## 
## Slot &quot;name&quot;:
## [1] &quot;Lisa&quot;
## 
## Slot &quot;age&quot;:
## [1] 36</code></pre>
<p>Example class: SummarizedExperiment</p>
<p>Representation:</p>
<p><code>RangedSummarizedExperiment</code> is an extension of
<code>SummarizedExperiment</code>. It uses <code>GRanges</code> to
define the genomic regions.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span></code></pre></div>
<pre><code>## Loading required package: MatrixGenerics</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
##     colWeightedMeans, colWeightedMedians, colWeightedSds,
##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
##     rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,
##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
##     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
##     table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## 
## Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:grDevices&#39;:
## 
##     windows</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## 
## Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MatrixGenerics&#39;:
## 
##     rowMedians</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     anyMissing, rowMedians</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;airway_small&quot;</span>,  <span class="at">package=</span><span class="st">&quot;BasicR&quot;</span>)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> airway_small</span></code></pre></div>
<div id="accessors" class="section level2">
<h2>Accessors</h2>
<p>To know all this, look for the help and the vignette.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(se)</span></code></pre></div>
<pre><code>## DataFrame with 8 rows and 9 columns
##            SampleName     cell      dex    albut        Run avgLength
##              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;
## SRR1039508 GSM1275862  N61311     untrt    untrt SRR1039508       126
## SRR1039509 GSM1275863  N61311     trt      untrt SRR1039509       126
## SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126
## SRR1039513 GSM1275867  N052611    trt      untrt SRR1039513        87
## SRR1039516 GSM1275870  N080611    untrt    untrt SRR1039516       120
## SRR1039517 GSM1275871  N080611    trt      untrt SRR1039517       126
## SRR1039520 GSM1275874  N061011    untrt    untrt SRR1039520       101
## SRR1039521 GSM1275875  N061011    trt      untrt SRR1039521        98
##            Experiment    Sample    BioSample
##              &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;
## SRR1039508  SRX384345 SRS508568 SAMN02422669
## SRR1039509  SRX384346 SRS508567 SAMN02422675
## SRR1039512  SRX384349 SRS508571 SAMN02422678
## SRR1039513  SRX384350 SRS508572 SAMN02422670
## SRR1039516  SRX384353 SRS508575 SAMN02422682
## SRR1039517  SRX384354 SRS508576 SAMN02422673
## SRR1039520  SRX384357 SRS508579 SAMN02422683
## SRR1039521  SRX384358 SRS508580 SAMN02422677</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(se)</span></code></pre></div>
<pre><code>## DataFrame with 4001 rows and 0 columns</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co">#assay(se)</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rowRanges</span>(se)</span></code></pre></div>
<pre><code>## GRangesList object of length 4001:
## $ENSG00000000003
## GRanges object with 17 ranges and 2 metadata columns:
##        seqnames            ranges strand |   exon_id       exon_name
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;     &lt;character&gt;
##    [1]        X 99883667-99884983      - |    667145 ENSE00001459322
##    [2]        X 99885756-99885863      - |    667146 ENSE00000868868
##    [3]        X 99887482-99887565      - |    667147 ENSE00000401072
##    [4]        X 99887538-99887565      - |    667148 ENSE00001849132
##    [5]        X 99888402-99888536      - |    667149 ENSE00003554016
##    ...      ...               ...    ... .       ...             ...
##   [13]        X 99890555-99890743      - |    667156 ENSE00003512331
##   [14]        X 99891188-99891686      - |    667158 ENSE00001886883
##   [15]        X 99891605-99891803      - |    667159 ENSE00001855382
##   [16]        X 99891790-99892101      - |    667160 ENSE00001863395
##   [17]        X 99894942-99894988      - |    667161 ENSE00001828996
##   -------
##   seqinfo: 722 sequences (1 circular) from an unspecified genome
## 
## ...
## &lt;4000 more elements&gt;</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co">#assays(se)[[1]]</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">assays</span>(se)<span class="sc">$</span>counts)</span></code></pre></div>
<pre><code>##                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516
## ENSG00000000003        679        448        873        408       1138
## ENSG00000000005          0          0          0          0          0
## ENSG00000000419        467        515        621        365        587
## ENSG00000000457        260        211        263        164        245
## ENSG00000000460         60         55         40         35         78
## ENSG00000000938          0          0          2          0          1
##                 SRR1039517 SRR1039520 SRR1039521
## ENSG00000000003       1047        770        572
## ENSG00000000005          0          0          0
## ENSG00000000419        799        417        508
## ENSG00000000457        331        233        229
## ENSG00000000460         63         76         60
## ENSG00000000938          0          0          0</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">metadata</span>(airway_small))</span></code></pre></div>
<pre><code>## [[1]]
## Experiment data
##   Experimenter name: Himes BE 
##   Laboratory: NA 
##   Contact information:  
##   Title: RNA-Seq transcriptome profiling identifies CRISPLD2 as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells. 
##   URL: http://www.ncbi.nlm.nih.gov/pubmed/24926665 
##   PMIDs: 24926665 
## 
##   Abstract: A 226 word abstract is available. Use &#39;abstract&#39; method.</code></pre>
<p>You can access each data “natively”, and even change it, but you
shouldn’t do that. Why? It is easy to screw it up…</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>se<span class="sc">@</span>rowRanges</span></code></pre></div>
<pre><code>## GRangesList object of length 4001:
## $ENSG00000000003
## GRanges object with 17 ranges and 2 metadata columns:
##        seqnames            ranges strand |   exon_id       exon_name
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;     &lt;character&gt;
##    [1]        X 99883667-99884983      - |    667145 ENSE00001459322
##    [2]        X 99885756-99885863      - |    667146 ENSE00000868868
##    [3]        X 99887482-99887565      - |    667147 ENSE00000401072
##    [4]        X 99887538-99887565      - |    667148 ENSE00001849132
##    [5]        X 99888402-99888536      - |    667149 ENSE00003554016
##    ...      ...               ...    ... .       ...             ...
##   [13]        X 99890555-99890743      - |    667156 ENSE00003512331
##   [14]        X 99891188-99891686      - |    667158 ENSE00001886883
##   [15]        X 99891605-99891803      - |    667159 ENSE00001855382
##   [16]        X 99891790-99892101      - |    667160 ENSE00001863395
##   [17]        X 99894942-99894988      - |    667161 ENSE00001828996
##   -------
##   seqinfo: 722 sequences (1 circular) from an unspecified genome
## 
## ...
## &lt;4000 more elements&gt;</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>se<span class="sc">@</span>metadata</span></code></pre></div>
<pre><code>## [[1]]
## Experiment data
##   Experimenter name: Himes BE 
##   Laboratory: NA 
##   Contact information:  
##   Title: RNA-Seq transcriptome profiling identifies CRISPLD2 as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells. 
##   URL: http://www.ncbi.nlm.nih.gov/pubmed/24926665 
##   PMIDs: 24926665 
## 
##   Abstract: A 226 word abstract is available. Use &#39;abstract&#39; method.</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="do">###don&#39;t do this:</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>se<span class="sc">@</span>rowRanges <span class="ot">&lt;-</span> se<span class="sc">@</span>rowRanges[<span class="dv">1</span><span class="sc">:</span><span class="dv">3000</span>]</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 4001 8 
## metadata(1): &#39;&#39;
## assays(1): counts
## rownames(3000): ENSG00000000003 ENSG00000000005 ... ENSG00000162951
##   ENSG00000162959
## rowData names(0):
## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>se<span class="sc">@</span>rowRanges</span></code></pre></div>
<pre><code>## GRangesList object of length 3000:
## $ENSG00000000003
## GRanges object with 17 ranges and 2 metadata columns:
##        seqnames            ranges strand |   exon_id       exon_name
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;     &lt;character&gt;
##    [1]        X 99883667-99884983      - |    667145 ENSE00001459322
##    [2]        X 99885756-99885863      - |    667146 ENSE00000868868
##    [3]        X 99887482-99887565      - |    667147 ENSE00000401072
##    [4]        X 99887538-99887565      - |    667148 ENSE00001849132
##    [5]        X 99888402-99888536      - |    667149 ENSE00003554016
##    ...      ...               ...    ... .       ...             ...
##   [13]        X 99890555-99890743      - |    667156 ENSE00003512331
##   [14]        X 99891188-99891686      - |    667158 ENSE00001886883
##   [15]        X 99891605-99891803      - |    667159 ENSE00001855382
##   [16]        X 99891790-99892101      - |    667160 ENSE00001863395
##   [17]        X 99894942-99894988      - |    667161 ENSE00001828996
##   -------
##   seqinfo: 722 sequences (1 circular) from an unspecified genome
## 
## ...
## &lt;2999 more elements&gt;</code></pre>
</div>
<div id="methods" class="section level2">
<h2>Methods</h2>
<div id="subsetting" class="section level3">
<h3>subsetting</h3>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># subsetting is very easy, just like with a data.frame</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> se[<span class="dv">1</span><span class="sc">:</span><span class="dv">2000</span>,]</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> se[,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 2000 3 
## metadata(1): &#39;&#39;
## assays(1): counts
## rownames(2000): ENSG00000000003 ENSG00000000005 ... ENSG00000092758
##   ENSG00000092820
## rowData names(0):
## colnames(3): SRR1039508 SRR1039509 SRR1039512
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
</div>
<div id="combining" class="section level3">
<h3>Combining</h3>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>rowbinded <span class="ot">&lt;-</span> <span class="fu">rbind</span>(se, se)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>rowbinded</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 4000 3 
## metadata(2): &#39;&#39; &#39;&#39;
## assays(1): counts
## rownames(4000): ENSG00000000003 ENSG00000000005 ... ENSG00000092758
##   ENSG00000092820
## rowData names(0):
## colnames(3): SRR1039508 SRR1039509 SRR1039512
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>colbinded <span class="ot">&lt;-</span> <span class="fu">cbind</span>(se, se)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>colbinded</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 2000 6 
## metadata(2): &#39;&#39; &#39;&#39;
## assays(1): counts
## rownames(2000): ENSG00000000003 ENSG00000000005 ... ENSG00000092758
##   ENSG00000092820
## rowData names(0):
## colnames(6): SRR1039508 SRR1039509 ... SRR1039509 SRR1039512
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>roi <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames=</span><span class="st">&quot;X&quot;</span>, <span class="at">ranges=</span><span class="dv">99800000</span><span class="sc">:</span><span class="dv">99850000</span>)</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>se_1 <span class="ot">&lt;-</span> <span class="fu">subsetByOverlaps</span>(se, roi)</span></code></pre></div>
</div>
</div>
<div id="granges-and-grangeslist" class="section level2">
<h2>Granges and GRangesList</h2>
<p><code>GRanges</code> objects store information about Genomic ranges
with chromosome, position and additional information. Offers a wide
range of functionality. <code>GRangesList</code> is a “grouped” version
of <code>GRnages</code>, it is similar to a normal list, but with
additional functionality. Both of them are able to handle strand
information as well.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>gr1 <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(</span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">seqnames =</span> <span class="st">&quot;chr2&quot;</span>,</span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="dv">103</span>, <span class="dv">106</span>),</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">strand =</span> <span class="st">&quot;+&quot;</span>,</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">score =</span> 5L, <span class="at">GC =</span> <span class="fl">0.45</span>)</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>gr2 <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">seqnames =</span> <span class="fu">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr3&quot;</span>),</span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="fu">c</span>(<span class="dv">107</span>, <span class="dv">113</span>), <span class="at">width =</span> <span class="dv">3</span>),</span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">strand =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>, <span class="st">&quot;-&quot;</span>),</span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">score =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>, <span class="at">GC =</span> <span class="fu">c</span>(<span class="fl">0.3</span>, <span class="fl">0.5</span>))</span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a>gr3 <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="st">&quot;chr2:102-107&quot;</span>)</span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a>gr4 <span class="ot">&lt;-</span> <span class="fu">makeGRangesFromDataFrame</span>(<span class="fu">data.frame</span>(<span class="at">chr=</span><span class="fu">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr2&quot;</span>), </span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">start=</span><span class="fu">c</span>(<span class="dv">104</span>, <span class="dv">104</span>), <span class="at">end=</span><span class="fu">c</span>(<span class="dv">108</span>, <span class="dv">105</span>),</span>
<span id="cb55-17"><a href="#cb55-17" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">name=</span><span class="fu">c</span>(<span class="st">&quot;gene1&quot;</span>, <span class="st">&quot;gene2&quot;</span>)), <span class="at">keep.extra.columns =</span> T, <span class="at">ignore.strand =</span> T)</span>
<span id="cb55-18"><a href="#cb55-18" aria-hidden="true" tabindex="-1"></a>grl <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>(<span class="st">&quot;txA&quot;</span> <span class="ot">=</span> gr1, <span class="st">&quot;txB&quot;</span> <span class="ot">=</span> gr2)</span>
<span id="cb55-19"><a href="#cb55-19" aria-hidden="true" tabindex="-1"></a>grl</span></code></pre></div>
<pre><code>## GRangesList object of length 2:
## $txA
## GRanges object with 1 range and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr2   103-106      + |         5      0.45
##   -------
##   seqinfo: 3 sequences from an unspecified genome; no seqlengths
## 
## $txB
## GRanges object with 2 ranges and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr1   107-109      + |         3       0.3
##   [2]     chr3   113-115      - |         4       0.5
##   -------
##   seqinfo: 3 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co">#seqinfo(gr1) &lt;- Seqinfo(genome=&quot;hg38&quot;)</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="co">#can have names for each row</span></span></code></pre></div>
<div id="accessors-1" class="section level3">
<h3>Accessors</h3>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seqnames</span>(gr1)</span></code></pre></div>
<pre><code>## factor-Rle of length 1 with 1 run
##   Lengths:    1
##   Values : chr2
## Levels(1): chr2</code></pre>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevels</span>(gr1)</span></code></pre></div>
<pre><code>## [1] &quot;chr2&quot;</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ranges</span>(gr1)</span></code></pre></div>
<pre><code>## IRanges object with 1 range and 0 metadata columns:
##           start       end     width
##       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
##   [1]       103       106         4</code></pre>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">start</span>(gr1)</span></code></pre></div>
<pre><code>## [1] 103</code></pre>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">end</span>(gr1)</span></code></pre></div>
<pre><code>## [1] 106</code></pre>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">strand</span>(gr1)</span></code></pre></div>
<pre><code>## factor-Rle of length 1 with 1 run
##   Lengths: 1
##   Values : +
## Levels(3): + - *</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">width</span>(gr1)</span></code></pre></div>
<pre><code>## [1] 4</code></pre>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(gr1)</span></code></pre></div>
<pre><code>## [1] 1</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(gr4)</span></code></pre></div>
<pre><code>## DataFrame with 2 rows and 1 column
##          name
##   &lt;character&gt;
## 1       gene1
## 2       gene2</code></pre>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(gr4)<span class="sc">$</span>name</span></code></pre></div>
<pre><code>## [1] &quot;gene1&quot; &quot;gene2&quot;</code></pre>
</div>
<div id="subsetting-1" class="section level3">
<h3>Subsetting</h3>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co">#data.frame-like</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>gr1[<span class="dv">1</span>,]</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr2   103-106      + |         5      0.45
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>gr4[<span class="dv">1</span>, <span class="st">&quot;name&quot;</span>]</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 1 metadata column:
##       seqnames    ranges strand |        name
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##   [1]     chr1   104-108      * |       gene1
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># subset function</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(gr1, strand <span class="sc">==</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr2   103-106      + |         5      0.45
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="splitting-and-combining" class="section level3">
<h3>Splitting and combining</h3>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>sp <span class="ot">&lt;-</span> <span class="fu">split</span>(gr2, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="fu">split</span>(gr4, <span class="sc">~</span> name)</span></code></pre></div>
<pre><code>## GRangesList object of length 2:
## $gene1
## GRanges object with 1 range and 1 metadata column:
##       seqnames    ranges strand |        name
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##   [1]     chr1   104-108      * |       gene1
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths
## 
## $gene2
## GRanges object with 1 range and 1 metadata column:
##       seqnames    ranges strand |        name
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##   [1]     chr2   104-105      * |       gene2
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(sp[[<span class="dv">1</span>]], sp[[<span class="dv">2</span>]])</span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr1   107-109      + |         3       0.3
##   [2]     chr3   113-115      - |         4       0.5
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">stack</span>(sp, <span class="at">index.var=</span><span class="st">&quot;name&quot;</span>)</span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 3 metadata columns:
##       seqnames    ranges strand |  name     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;Rle&gt; &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr1   107-109      + |     1         3       0.3
##   [2]     chr3   113-115      - |     2         4       0.5
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="aggregating" class="section level3">
<h3>Aggregating</h3>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">aggregate</span>(gr2, score <span class="sc">~</span> strand, mean)</span></code></pre></div>
<pre><code>## DataFrame with 2 rows and 2 columns
##     strand     score
##   &lt;factor&gt; &lt;numeric&gt;
## 1        +         3
## 2        -         4</code></pre>
</div>
<div id="interval-operations" class="section level3">
<h3>Interval operations</h3>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">flank</span>(gr1, <span class="dv">10</span>, <span class="at">both=</span>T)</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr2    93-112      + |         5      0.45
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">shift</span>(gr2, <span class="dv">5</span>)</span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr1   112-114      + |         3       0.3
##   [2]     chr3   118-120      - |         4       0.5
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">resize</span>(gr1, <span class="dv">20</span>)</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr2   103-122      + |         5      0.45
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(gr1, gr3)</span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr2   103-106      + |         5      0.45
##   [2]     chr2   102-107      * |      &lt;NA&gt;        NA
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb100"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a> <span class="fu">reduce</span>(<span class="fu">c</span>(gr1, gr3)) <span class="co"># combines the overlapping ranges</span></span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr2   103-106      +
##   [2]     chr2   102-107      *
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb102"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a> <span class="fu">reduce</span>(<span class="fu">c</span>(gr1, gr3), <span class="at">ignore.strand=</span>T)</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr2   102-107      *
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="comapring-2-granges" class="section level3">
<h3>Comapring 2 GRanges</h3>
<div class="sourceCode" id="cb104"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co">#union(gr2, gr4)</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a><span class="co">#intersect(gr2, gr4)</span></span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a><span class="co">#setdiff(gr2, gr4)</span></span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a><span class="co">#hits &lt;- findOverlaps(gr2, gr4, ignore.strand=TRUE)</span></span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a><span class="co">#countOverlaps(gr2, gr4, ignore.strand=TRUE)</span></span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a><span class="co">#subsetByOverlaps(gr2, gr4)</span></span></code></pre></div>
</div>
</div>
<div id="deseq2" class="section level2">
<h2>DeSeq2</h2>
<div class="sourceCode" id="cb105"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;pasilla&quot;</span>)</span></code></pre></div>
<pre><code>## Loading required package: DEXSeq</code></pre>
<pre><code>## Loading required package: BiocParallel</code></pre>
<pre><code>## Loading required package: DESeq2</code></pre>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## Loading required package: RColorBrewer</code></pre>
<div class="sourceCode" id="cb111"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>pasCts <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;pasilla_gene_counts.tsv&quot;</span>,</span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">package=</span><span class="st">&quot;pasilla&quot;</span>, <span class="at">mustWork=</span><span class="cn">TRUE</span>)</span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>pasAnno <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,</span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;pasilla_sample_annotation.csv&quot;</span>,</span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">package=</span><span class="st">&quot;pasilla&quot;</span>, <span class="at">mustWork=</span><span class="cn">TRUE</span>)</span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">read.csv</span>(pasCts,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span><span class="st">&quot;gene_id&quot;</span>))</span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(pasAnno, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> coldata[,<span class="fu">c</span>(<span class="st">&quot;condition&quot;</span>,<span class="st">&quot;type&quot;</span>)]</span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-11"><a href="#cb111-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(coldata) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;fb&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(coldata))</span>
<span id="cb111-12"><a href="#cb111-12" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> cts[, <span class="fu">rownames</span>(coldata)]</span>
<span id="cb111-13"><a href="#cb111-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-14"><a href="#cb111-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;DESeq2&quot;</span>)</span>
<span id="cb111-15"><a href="#cb111-15" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb111-16"><a href="#cb111-16" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb111-17"><a href="#cb111-17" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> condition)</span></code></pre></div>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<div class="sourceCode" id="cb113"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>dds</span></code></pre></div>
<pre><code>## class: DESeqDataSet 
## dim: 14599 7 
## metadata(1): version
## assays(1): counts
## rownames(14599): FBgn0000003 FBgn0000008 ... FBgn0261574 FBgn0261575
## rowData names(0):
## colnames(7): treated1 treated2 ... untreated3 untreated4
## colData names(2): condition type</code></pre>
<div class="sourceCode" id="cb115"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[<span class="dv">1</span><span class="sc">:</span><span class="dv">200</span>,]</span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb122"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>res</span></code></pre></div>
<pre><code>## log2 fold change (MLE): condition untreated vs treated 
## Wald test p-value: condition untreated vs treated 
## DataFrame with 200 rows and 6 columns
##                baseMean log2FoldChange     lfcSE       stat      pvalue
##               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;   &lt;numeric&gt;
## FBgn0000003    0.172207     -0.9986796  3.806863 -0.2623366   0.7930619
## FBgn0000008   95.873798      0.0140036  0.219868  0.0636906   0.9492166
## FBgn0000014    1.090634      0.5233543  2.215680  0.2362048   0.8132737
## FBgn0000015    0.853818      1.8909340  2.149959  0.8795209   0.3791189
## FBgn0000017 4392.045082      0.2549107  0.116978  2.1791389   0.0293213
## ...                 ...            ...       ...        ...         ...
## FBgn0000567   263.99287       1.239187  0.166174   7.457181 8.83932e-14
## FBgn0000568  3655.58621       0.220186  0.210651   1.045263 2.95901e-01
## FBgn0000575  1435.99029      -0.300818  0.175862  -1.710531 8.71677e-02
## FBgn0000576     0.00000             NA        NA         NA          NA
## FBgn0000577     0.35367       1.822069  3.491008   0.521932 6.01718e-01
##                    padj
##               &lt;numeric&gt;
## FBgn0000003    0.994466
## FBgn0000008    0.996580
## FBgn0000014    0.994466
## FBgn0000015    0.817888
## FBgn0000017    0.189072
## ...                 ...
## FBgn0000567 8.26476e-12
## FBgn0000568 7.88180e-01
## FBgn0000575 3.97570e-01
## FBgn0000576          NA
## FBgn0000577 9.94466e-01</code></pre>
<p>```</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
      </div>

</div>



      <footer><div class="copyright">
  <p></p><p>Developed by Reka Toth, Rajbir Nath Batra.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>
